### PR TITLE
[OpenVINO] Implements LSTM and GRU, fixes RNN shape bug

### PR DESCRIPTION
**Pull request by goyaladitya05**
This PR implements `lstm()` and `gru()` for the OpenVINO backend using `ov_opset.lstm_sequence` and `gru_sequence`, replacing the `backend.rnn()` fallback for the common case.

This PR also fixes a pre-existing shape bug in `backend.rnn()` that caused all RNN-based layers (LSTM, GRU, SimpleRNN, StackedRNN) to fail or produce wrong results with `implementation=2` (the default).

**Fix:** Declared body params as `[1, batch, features]`, squeeze axis 0 before calling `step_function`, and unsqueeze the step output back to `[1, batch, units]` before `get_concatenated_slices`.

### `LSTM` and `GRU` implementation

Uses `ov_opset.lstm_sequence`, `gru_sequence` when all of the following hold:
- activations are tanh + sigmoid
- unroll=False
- bias is provided
- mask is None 
- GRU only: `reset_after=True`

Weight layout differences handled:
- LSTM: Keras gate order `[i, f, c, o]` vs OV `[f, i, c, o]` (reordered via split/concat)
- GRU: same gate order `[z, r, h]` in both; bias `[2, 3*units]` vs OV `[1, 4*units]` = `[b_z+rb_z, b_r+rb_r, b_h, rb_h]` for `linear_before_reset=True`

Removed `numpy_scan` from rnn.py. It was implemented to throw `NotImplementedError`, and was not present at all for other backends (Tensorflow, Torch, JAX).

Removed `unstack` from rnn.py. Tensorflow and Torch have it being implemented in core.py, and not in rnn.py. 

### Tests
Enabled 30 tests releted to RNN,LSTM and GRU from `excluded_concrete_tests.txt`

Closes: #22435, Closes: openvinotoolkit/openvino/issues/34403, Closes: openvinotoolkit/openvino/issues/34408

### Note
After this PR, rnn.py is at parity with other backends.  `RNN`, `LSTM`, `GRU`, and `cudnn_ok` are all implemented.

### CODE REVIEW

File: keras/src/backend/openvino/rnn.py [hunk 1]
```diff
@@ -284,12 +293,225 @@ def _slice_at_0(x):
...
+    ).output(0)
 
-def gru(*args, **kwargs):
-    raise NotImplementedError("`gru` is not supported with openvino backend")
+    b = _reorder_gates(
+        bias_ov, ["i", "f", "c", "o"], ["f", "i", "c", "o"], axis=0
+    )
+    b = ov_opset.unsqueeze(
+        b, ov_opset.constant([0], dtype=Type.i32).output(0)
+    ).output(0)
```
[the remark this answers is not in the document]
Comment (goyaladitya05) [reply keras/src/backend/openvino/rnn.py:384]: The Keras LSTMCell uses a single combined bias b added once to z = Wx + Rh + b, there are no separate Wb and Rb. We place this bias in the Wb position; OV opset15 accepts [1, 4*hidden_size] and treats absent Rb as zeros. Adding explicit zero Rb would produce the same result and is unnecessary.

File: keras/src/backend/openvino/excluded_concrete_tests.txt [hunk 2]
```diff
@@ -396,12 +371,7 @@ SparseCategoricalCrossentropyTest::test_sample_weighted
 SparseCategoricalCrossentropyTest::test_scalar_weighted
 SparseCategoricalCrossentropyTest::test_unweighted
 SpectralNormalizationTest::test_apply_layer
-StackedRNNTest::test_correctness_single_state_stack
-StackedRNNTest::test_correctness_two_states_stack
 StackedRNNTest::test_return_state_stacked_lstm_cell
```
Comment (goyaladitya05) [inline keras/src/backend/openvino/excluded_concrete_tests.txt:374]: `StackedRNNTest::test_return_state_stacked_lstm_cell` can be enabled once `qr` op is implemented in linalg.py. It won't pass right now.

**Review by hertschuh — APPROVED** [review]
Thanks for adding all of these!
